internal/storage/references/registry_reference.go

```diff
@@ -24,16 +24,27 @@ type RegistryImageReference struct {
 // RegistryImageReferenceFromRaw is an internal constructor of a RegistryImageReference.
 //
 // This should only be called from internal/storage.
-// It’s the caller’s responsibility to provide a valid value (!IsNameOnly, and registry-qualified).
-func RegistryImageReferenceFromRaw(rawNamed reference.Named) RegistryImageReference {
+// It does two validations on rawNamed:
+// - It validates the image either has a tag or a digest, but not both (preferring digest)
+// - Verifies the image is not only a name.
+func RegistryImageReferenceFromRaw(rawNamed reference.Named) (RegistryImageReference, error) {
+	_, isTagged := rawNamed.(reference.NamedTagged)
+	canonical, isDigested := rawNamed.(reference.Canonical)
+	if isTagged && isDigested {
+		canonical, err := reference.WithDigest(reference.TrimNamed(rawNamed), canonical.Digest())
+		if err != nil {
+			return RegistryImageReference{}, err
```

Comment:
(The only way this can fail is if `canonical.Digest()` is invalid, and that shouldn’t have happened by induction, so I’d be fine with a `panic` here. But returning an error explicitly is perfectly valid as well.)

Reply:
that's better because then I don't need to change the function signature

Reply:
Also using `panic` does not leave around (supposedly) 100% unreachable error handling code paths, considering that CRI-O is measuring code coverage.